File: service/app/src/test/java/io/miragon/blueprint/domain/bike/BikeIdTest.java
package io.miragon.blueprint.domain.bike;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatThrownBy;

import org.junit.jupiter.api.Test;

class BikeIdTest {

    @Test
    void exposesTheWrappedBikeId() {
        // given/when: a bike id is created from a non-blank value
        BikeId bikeId = new BikeId("BIKE-900");
        // then: the raw value is exposed unchanged
        assertThat(bikeId.value()).isEqualTo("BIKE-900");
    }

    @Test
    void rejectsABlankBikeId() {
        // when/then: a blank value is refused
        assertThatThrownBy(() -> new BikeId("   "))
                .isInstanceOf(IllegalArgumentException.class);
    }
}

File: service/app/src/test/kotlin/io/miragon/blueprint/domain/bike/BikeIdTest.kt
package io.miragon.blueprint.domain.bike

import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.api.Assertions.assertThatThrownBy
import org.junit.jupiter.api.Test

class BikeIdTest {

    @Test
    fun `exposes the wrapped bike id`() {
        // given/when: a bike id is created from a non-blank value
        val bikeId = BikeId("BIKE-900")
        // then: the raw value is exposed unchanged
        assertThat(bikeId.value).isEqualTo("BIKE-900")
    }

    @Test
    fun `rejects a blank bike id`() {
        // when/then: a blank value is refused
        assertThatThrownBy { BikeId("   ") }
            .isInstanceOf(IllegalArgumentException::class.java)
    }
}
